ga4gh/server/datamodel/variants.py

```diff
@@ -37,6 +38,34 @@ def isUnspecified(str):
     return str == "" or str is None
 
 
+def deepGetAttr(obj, path):
+    """
+    Utility function which is similiar to std lib's getattr, but works
+    to resolve a dot-delimited path on an object. If path is not found
+    an `AttributeError` will be raised.
+    """
+    return reduce(getattr, path.split('.'), obj)
+
+
+def deepSetAttr(obj, path, val):
```

Comment:
These could probably go into protocol in schemas. I think this pattern of dot access for dictionaries is a sane way to work with data when you don't want to make a new class in python, and mocks the protocol buffers and peewee models well.

This is what I did for the peer service
```
class Struct:
    def __init__(self, **entries):
        self.__dict__.update(entries)

###

map(lambda x: Struct(**x), self._peers[offset:offset + limit])
```
https://github.com/ga4gh/server/pull/1556/files#diff-02d20abc7e4d68c8ae177064ca45176aR49